golang/cosmos/app/app.go

```diff
@@ -788,6 +767,41 @@ func NewAgoricApp(
 	return app
 }
 
+func upgrade8Handler(app *GaiaApp, upgradeName string) func(ctx sdk.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+	return func(ctx sdk.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		//Run migrations first so InitGenesis is called for swingset, vibc, vbank, vstorage, lien
+		vm, err := app.mm.RunMigrations(ctx, app.configurator, vm)
+		if err != nil {
+			return vm, err
+		}
+
+		// Set bean count
+		currentSwingsetParams := app.SwingSetKeeper.GetParams(ctx)
+		ctx.Logger().Info("pre-swingset upgrade", "subspace", swingset.ModuleName, "params", currentSwingsetParams)
+
+		for i := 0; i < len(currentSwingsetParams.BeansPerUnit); i++ {
+			if currentSwingsetParams.BeansPerUnit[i].Key == swingsettypes.BeansPerBlockComputeLimit {
+				currentSwingsetParams.BeansPerUnit[i].Beans = sdk.NewUint(6500000000)
+			}
+		}
+
+		// Set bootstrap
+		switch upgradeName {
+		case upgradeName:
+			currentSwingsetParams.BootstrapVatConfig = "@agoric/vats/decentral-main-psm-config.json"
+		case upgradeNameTest:
+			currentSwingsetParams.BootstrapVatConfig = "@agoric/vats/decentral-test-psm-config.json"
+		default:
+			return vm, fmt.Errorf("invalid upgrade name")
+		}
```

Comment:
Consider switching to compute the config file name and consolidating thereafter:

```
var configPath string
switch name {
case upgradeName:
        configPath = "@agoric/vats/decentral-main-psm-config.json"
case upgradeNameTest:
        configPath = "@agoric/vats/decentral-test-psm-config.json"
default:
        return vm, fmt.Errorf("invalid upgrade name: %s", name)
}
currentSwingsetParams.BootstrapVatConfig = configPath
```